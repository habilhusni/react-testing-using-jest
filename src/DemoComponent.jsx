import React from "react";

export default class DemoComponent extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: "husni"
    };
  }

  handleClick = () => {
    this.setState({
      name: "kevin"
    });
  };

  componentDidMount() {
    this.setState({
      name: "herry"
    });
  }

  render() {
    return <div onClick={this.handleClick}>Hello hai {this.state.name}</div>;
  }
}
