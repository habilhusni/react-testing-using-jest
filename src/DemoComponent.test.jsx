import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";

import DemoComponent from "./DemoComponent";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;

  beforeEach(() => {
    component = shallow(<DemoComponent />);
  });

  it("Testing DemoComponent Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("Testing handleClick", () => {
    component.find("div").simulate("click");
    expect(component.state("name")).toEqual("kevin");
  });
});

it("Testing ComponentDidMount", () => {
  let component = shallow(<DemoComponent />);
  component.instance().componentDidMount();
  expect(component.state("name")).toEqual("habil");
});
