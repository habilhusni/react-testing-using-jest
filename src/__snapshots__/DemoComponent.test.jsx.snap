// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing DemoComponent Snapshot 1`] = `
<div
  onClick={[Function]}
>
  Hello hai 
  herry
</div>
`;
